app/routes/_layout+/kits.$kitId.manage-assets.tsx

```diff
@@ -0,0 +1,510 @@
+import { useEffect, useMemo } from "react";
+import { AssetStatus, type Prisma } from "@prisma/client";
+import { json, redirect } from "@remix-run/node";
+import type { ActionFunctionArgs, LoaderFunctionArgs } from "@remix-run/node";
+import { Form, useLoaderData, useNavigation } from "@remix-run/react";
+import { useAtom, useAtomValue } from "jotai";
+import { z } from "zod";
+import { kitsSelectedAssetsAtom } from "~/atoms/selected-assets-atoms";
+import { AssetImage } from "~/components/assets/asset-image";
+import DynamicDropdown from "~/components/dynamic-dropdown/dynamic-dropdown";
+import { FakeCheckbox } from "~/components/forms/fake-checkbox";
+import { ChevronRight } from "~/components/icons/library";
+import Header from "~/components/layout/header";
+import { List } from "~/components/list";
+import { Filters } from "~/components/list/filters";
+import { Button } from "~/components/shared/button";
+import { Image } from "~/components/shared/image";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "~/components/shared/tooltip";
+import { Td } from "~/components/table";
+import { db } from "~/database/db.server";
+import {
+  createBulkKitChangeNotes,
+  createNote,
+  getPaginatedAndFilterableAssets,
+} from "~/modules/asset/service.server";
+import { getUserByID } from "~/modules/user/service.server";
+import { makeShelfError, ShelfError } from "~/utils/error";
+import { isFormProcessing } from "~/utils/form";
+import { data, error, getParams, parseData } from "~/utils/http.server";
+import {
+  PermissionAction,
+  PermissionEntity,
+} from "~/utils/permissions/permission.validator.server";
+import { requirePermission } from "~/utils/roles.server";
+import { tw } from "~/utils/tw";
+
+export async function loader({ context, request, params }: LoaderFunctionArgs) {
+  const authSession = context.getSession();
+  const { userId } = authSession;
+
+  const { kitId } = getParams(params, z.object({ kitId: z.string() }), {
+    additionalData: { userId },
+  });
+
+  try {
+    const { organizationId } = await requirePermission({
+      userId,
+      request,
+      entity: PermissionEntity.kit,
+      action: PermissionAction.update,
+    });
+
+    const [kit, assets] = await Promise.all([
```

Comment:
I think its confusing here to call the second item `assets` because its more than that. I would call it `assetsData` or something like this.